### PR TITLE
Check observer existence in domobserver methods

### PR DESCRIPTION
**Pull request by ranmocy**

### CODE REVIEW

**Comment by ranmocy** [comment]
Updated for flush().

Not in production, but in tests with jsdom framework. It's undefined function issue since `this.observer.takeRecords()` is called but `this.observer` is undefined.

jsdom added the support of mutationobserver in https://github.com/jsdom/jsdom/issues/639, and I'm waiting for next version of jest to include the [version bump](https://github.com/facebook/jest/pull/8851).

Even though it will eventually work, I still think it's better to do so for correctness. The observer should either be nullable in the caller, or silently no-op when methods are called.

File: src/domobserver.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ export class DOMObserver {
   }
 
   flush() {
-    if (!this.view.docView || this.flushingSoon) return
+    if (!this.view.docView || this.flushingSoon || !this.observer) return
```
Comment (marijnh) [inline src/domobserver.js]: I think it'd make more sense to put the check in the next statement—it's possible for there to be events on the queue (via the chardata event handler) even when there isn't a mutation observer, and selection changes are also flushed through here.

Comment (ranmocy) [reply src/domobserver.js]: Done.

**Comment by marijnh** [comment]
Thank you! Merged as f329f82

**Pull request closed without merging**